daemon/docker.go

```diff
@@ -91,8 +91,11 @@ func (s *daemon) createDockerVolume(request *pluginRequest) (*Volume, error) {
 
 func (s *daemon) getDockerVolume(r *http.Request) (*Volume, *pluginRequest, error) {
 	request, err := convertToPluginRequest(r)
+	if err != nil {
```

Comment:
Fail fast and add some extra info. The next line could have thrown a null pointer exception when trying to access Opts if request was nil (which it can be).